### PR TITLE
Monorepo Everything

### PR DESCRIPTION
**Pull request by Conner1115**
Mono-repoifying react and the normal extension lib

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+  - [React Client](https://docs.replit.com/extensions/category/react)
+- [Discourse Category](https://ask.replit.com/c/extensions)
 - [React Extension Template](https://replit.com/@replit/React-Extension?v=1)
 - [HTML/CSS/JS Extension Template](https://replit.com/@replit/HTMLCSSJS-Extension?v=1)
 
-[![Run on Replit button](https://user-images.githubusercontent.com/50180265/228865994-ccf7348e-ffb7-454e-bc4e-ce90df6c09bc.png)](https://replit.com/github/replit/extensions)
-
-## Help
-If you don't understand something in the documentation, have found a bug, or would like to request a feature, you can get help on the [Ask Forum](https://ask.replit.com/c/extensions).
+## Developer Guide
```
Comment (lunaroyster) [inline README.md:17]: I guess each of the packages should have their own developer guides

Comment (Conner1115) [reply README.md:17]: I'll write some more detailed stuff on it

File: .replit [hunk 2]
```diff
@@ -1,4 +1,4 @@
-run = "npm run dev"
+run = "cd extensions && npm run dev"
```
Comment (lunaroyster) [inline .replit:1]: don't we need to `cd` to modules/example instead?

```suggestion
run = "cd modules/example && npm run dev"
```

File: package.json [hunk 3]
```diff
@@ -1,46 +1,39 @@
...
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "build": "rm -rf dist && tsup src/index.ts  --sourcemap --dts --format esm,cjs,iife --global-name replit",
-    "lint": "npx prettier --write src/* && cd example && npx prettier --write .",
-    "lint:check": "npx prettier -l src/* && cd example && npx prettier -l .",
-    "type:check": "tsc --noEmit",
-    "dev": "npm run build && cd ./example && npm install && npm run dev",
-    "prepublishOnly": "npm run build",
-    "generate:docs": "npx typedoc src/index.ts --json docs/main.json --plugin ./util/signature-plugin/index.cjs"
+    "dev": "cd modules/example && pnpm dev",
```
Comment (lunaroyster) [inline package.json:5]: I think this script should also build the extensions package, or is that no longer required?

**Review by lunaroyster — APPROVED** [review]
Okay, i think we're ready! @Conner1115 thanks for leading the effort!

- There were some issues with typescript in the monorepo that should now be fixed.

- The developer flow is still the same. If we could somehow obviate the need for rebuilds and reload the libraries with HMR in the example app, that would be oh-so-incredible

- Unfortunately I wasn't able to get typescript aliases working reliably (they work perfectly within the scope of a package, but throw a bunch of errors when used cross-package), but we can follow up with that. 

- I dropped replit/extensions-react to 0.x.x since we're still iterating on it

**Pull request merged**